src/solvers/sat/satcheck_minisat2.cpp

```diff
@@ -140,8 +140,22 @@ void satcheck_minisat2_baset<T>::lcnf(const bvt &bv)
 
     solver->addClause_(c);
 
+    // To map clauses to lines of program code, track clause indices in the
```

Comment:
I don't want to be a downer on this but ...

1. There is not necessarily a 1-to-1 correlations between clauses and lines of code.  For example, rewriting and normalisation may mean that multiple lines reduce to the same shared term.  Which line should this be credited to?  This is worse when it is things like multiplication where additions from other operations can appear as sub-parts of the operator.

2. Number of variables and clauses is a *very* crude measure of SAT difficulty.  Reducing the number of clauses can make the problem harder; I have seen multiple instances of this.